prov/verbs/src/verbs_info.c

```diff
@@ -361,7 +362,12 @@ int fi_ibv_check_rx_attr(const struct fi_rx_attr *attr,
 		return -FI_ENODATA;
 	}
 
-	if (attr->total_buffered_recv > info->rx_attr->total_buffered_recv) {
+	rm_enabled =(info->domain_attr &&
+		     info->domain_attr->resource_mgmt == FI_RM_ENABLED);
+
+	if (!rm_enabled &&
+	    (attr->total_buffered_recv > info->rx_attr->total_buffered_recv))
+	{
```

Comment:
I believe the above check should be there even for FI_RM_ENABLED. If the user requests buffering of length > supported by the provider we have to return an error. In case of FI_RM_DISABLED, the provider's total_buffered_recv would always be zero. Sorry if I wasn't clear before. :)

So the original code before this PR's change was already correct. If verbs/RDM supports resource management, it should set the total_buffered_recv to the amount of buffering provided.